modules/product/commerce_product.module

```diff
@@ -71,6 +73,24 @@ function commerce_product_theme() {
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function commerce_product_commerce_product_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  /** @var \Drupal\commerce_product\Entity\ProductInterface $entity */
+
+  $product_type = ProductType::load($entity->bundle());
+  if ($product_type->shouldInjectVariationFields()) {
+    /** @var \Drupal\commerce_product\ProductVariationFieldRendererInterface $variation_field_injector */
+    $variation_field_injector = \Drupal::service('commerce_product.variation_field_renderer');
+
+    $injection_fields = $variation_field_injector->getRenderableFields($entity->getDefaultVariation(), $view_mode);
+    foreach ($injection_fields as $field_name => $injection_field) {
+      $build[$field_name] = $injection_field;
```

Comment:
We said we'd prefix the entries in $build with 'variation_' to prevent collisions and make it more obvious where the fields came from?